### PR TITLE
Add Team Workflow Deletions to Teams API

### PR DESCRIPTION
**Pull request by nsprenkle**
**Context:** Implement workflow deletion functions at team assignment API level.

**Changes:**
- Implement and test (API)
    - `cancel_workflow`
    - `get_assessment_workflow_cancellation`
    - `is_workflow_cancelled`
- Delete `TeamAssessmentWorkflowCancellation` model and serializer, inherit behavior from parent class
- Version bump to ~2.6.27~ ~2.6.28~ 2.6.29
- Add extra spacing in `setup.py` to remove pylint warning

JIRA: 1/2 of [EDUCATOR-4978](https://openedx.atlassian.net/browse/EDUCATOR-4978)
FYI: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/workflow/team_api.py [hunk 1]
```diff
@@ -155,3 +163,62 @@ def get_status_counts(course_id, item_id):
         for status, count
         in counts_by_status.items()
     ]
+
+
+def cancel_workflow(team_submission_uuid, comments, cancelled_by_id):
+    """
+    Add an entry in AssessmentWorkflowCancellation table for a AssessmentWorkflow.
```
Comment (Andytr1) [inline openassessment/workflow/team_api.py:170]: nit: Technically it's TeamAssessmentWorkflow that should be referenced, not AssessmentWorkflow.

File: openassessment/workflow/team_api.py [hunk 2]
```diff
@@ -155,3 +163,62 @@ def get_status_counts(course_id, item_id):
...
+        submission_uuid = _get_workflow_model(team_submission_uuid).submission_uuid
+        TeamAssessmentWorkflow.cancel_workflow(
+            submission_uuid,
+            comments,
+            cancelled_by_id,
+            TeamAssessmentWorkflow.REQUIREMENTS
+        )
+    except Exception as exc:
+        err_msg = (
+            u"Could not cancel team assessment workflow with team_submission_uuid {uuid} due to error: {exc}"
```
Comment (Andytr1) [inline openassessment/workflow/team_api.py:188]: nit: don't need 'u' prefix anymore as all python 3 strings are unicode.

File: setup.py [hunk 3]
```diff
@@ -33,9 +33,10 @@ def load_requirements(*requirements_paths):
             )
     return list(requirements)
 
+
```
Comment (jansenk) [inline setup.py:36]: nit: remove blank line

Comment (nsprenkle) [reply setup.py:36]: That was actually intentional: Pylint expects an extra line before the `setup` function so this removes a warning message.

**Pull request merged**